src/NHibernate/Linq/Functions/CompareGenerator.cs

```diff
@@ -44,9 +44,17 @@ internal static bool IsCompareMethod(MethodInfo methodInfo)
 				return true;
 
 			// This is .Net 4 only, and in the System.Data.Services assembly, which we don't depend directly on.
-			return methodInfo != null && methodInfo.Name == "Compare" &&
+			if (methodInfo != null && methodInfo.Name == "Compare" &&
 				   methodInfo.DeclaringType != null &&
-				   methodInfo.DeclaringType.FullName == "System.Data.Services.Providers.DataServiceProviderMethods";
+				   methodInfo.DeclaringType.FullName == "System.Data.Services.Providers.DataServiceProviderMethods")
+				return true;
+
+			if (methodInfo != null && methodInfo.Name == "CompareString" &&
```

Comment:
This is normally handled by `VBStringComparisonExpression`. The VB Core/New Roslyn case will be fixed in re-linq 2.2 (it's in alpha-3 now)